lib/sprockets/manifest.rb

```diff
@@ -179,31 +179,23 @@ def compile(*args)
         assets[asset.logical_path] = asset.digest_path
 
         target = File.join(dir, asset.digest_path)
+        compressor = create_compressor(asset, target, executor)
+        concurrent_compressors << compressor if compressor
 
         if File.exist?(target)
           logger.debug "Skipping #{target}, already exists"
+          compressor.execute if compressor
         else
           logger.info "Writing #{target}"
```

Comment:
I think we need to run `create_compressor` after each time we log since it outputs to logs right away Otherwise since `create_compressor` immediately outputs a log, you would could get something in your output like

```
Writing #{target}.gz
Writing #{target}
```

or 

```
"Skipping #{target}.gz, already exists"
"Skipping #{target}, already exists"
```

Which is backwards.